src/client/api/rpc.c

```diff
@@ -39,11 +39,6 @@ daos_rpc_send(crt_rpc_t *rpc, tse_task_t *task)
 	int rc;
 
 	rc = crt_req_send(rpc, daos_rpc_cb, task);
-	if (rc != 0) {
-		/** task will be completed in CB above */
-		rc = 0;
-	}
-
```

Comment:
since it provided daos_rpc_cb(), it will be called in failure case and inside daos_rpc_cb will set task's completion status. Here better to keep same as original code, to avoid duplicated err handling for some cases.

Reply:
Hmm. I need to review it again. I think this callback is not called in case of errors.

Reply:
Ahh. Yes, you right. It will be called in all cases. Will restore it.